blockchain/chain_sync/src/sync.rs

```diff
@@ -56,7 +56,7 @@ pub struct ChainSyncer<DB, TBeacon> {
     state: Arc<RwLock<SyncState>>,
 
     /// Drand randomness beacon
-    beacon: Arc<TBeacon>,
+    beacon: Arc<RwLock<TBeacon>>,
```

Comment:
I was considering exactly that, but since `fn verity_entry` is now expects `&mut self`, I had to replace `Arc<TBeacon>` with `Arc<RwLock<TBeacon>>`.

Reply:
But if you change the mutex to be around the cache, you won't need to change `verify_entry` to require a mutable reference to the beacon client. I made the change to make sure I wasn't missing anything, have a look at this commit if what I'm saying is still confusing (and feel free to cherry pick it if it's easier for you): https://github.com/ChainSafe/forest/commit/565172b9fa02299a6fd4181aaa1b119a021c5f47

Reply:
oh, that's very interesting, syntax checker is definitely happy with just `&self` now. It's either I got confused or it was an issue with `rls`; since then I switched to `rust-analyzer` and no error is present anymore.  

Thank you for your advice! This has now been updated.